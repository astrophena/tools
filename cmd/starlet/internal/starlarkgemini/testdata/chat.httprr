httprr trace v1
457 2243
POST https://generativelanguage.googleapis.com/v1beta/models/gemini-1.5-flash:generateContent HTTP/1.1
Host: generativelanguage.googleapis.com
User-Agent: starlarkgemini.test/ (+https://astrophena.name/bleep-bloop)
Content-Length: 184
Content-Type: application/json

{"contents":[{"parts":[{"text":"How are you?"}],"role":"user"},{"parts":[{"text":"I'm fine, thank you!"}],"role":"model"},{"parts":[{"text":"Tell me about yourself."}],"role":"user"}]}HTTP/2.0 200 OK
Alt-Svc: h3=":443"; ma=2592000,h3-29=":443"; ma=2592000
Cache-Control: private
Content-Type: application/json; charset=UTF-8
Date: Sun, 01 Sep 2024 10:55:15 GMT
Server: scaffolding on HTTPServer2
Server-Timing: gfet4t7; dur=1414
Vary: Origin
Vary: X-Origin
Vary: Referer
X-Content-Type-Options: nosniff
X-Frame-Options: SAMEORIGIN
X-Xss-Protection: 0

{
  "candidates": [
    {
      "content": {
        "parts": [
          {
            "text": "I am a large language model, trained by Google.\n\n**Here's what you need to know about me:**\n\n* **I'm a computer program:** I don't have feelings, emotions, or a physical body.\n* **I'm trained on a massive dataset:** I can generate text, translate languages, write different kinds of creative content, and answer your questions in an informative way. \n* **I'm constantly learning:** My abilities are constantly improving as I process more data and interact with users.\n* **I'm not a person:** I don't have personal experiences or opinions. \n* **I'm here to help:**  I can assist you with a wide range of tasks, from writing emails to summarizing articles.\n\n**What would you like to know about me specifically?** \n"
          }
        ],
        "role": "model"
      },
      "finishReason": "STOP",
      "index": 0,
      "safetyRatings": [
        {
          "category": "HARM_CATEGORY_SEXUALLY_EXPLICIT",
          "probability": "NEGLIGIBLE"
        },
        {
          "category": "HARM_CATEGORY_HATE_SPEECH",
          "probability": "NEGLIGIBLE"
        },
        {
          "category": "HARM_CATEGORY_HARASSMENT",
          "probability": "NEGLIGIBLE"
        },
        {
          "category": "HARM_CATEGORY_DANGEROUS_CONTENT",
          "probability": "NEGLIGIBLE"
        }
      ],
      "citationMetadata": {
        "citationSources": [
          {
            "startIndex": 218,
            "endIndex": 341,
            "uri": "https://www.nptechforgood.com/2023/04/23/how-to-enhance-your-nonprofits-written-content-with-artificial-intelligence/",
            "license": ""
          }
        ]
      }
    }
  ],
  "usageMetadata": {
    "promptTokenCount": 20,
    "candidatesTokenCount": 172,
    "totalTokenCount": 192
  }
}
