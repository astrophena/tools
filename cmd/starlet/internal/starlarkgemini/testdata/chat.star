result = gemini.generate_content(
    contents = ["How are you?", "I'm fine, thank you!", "Tell me about yourself."],
)

print(result[0][0])
