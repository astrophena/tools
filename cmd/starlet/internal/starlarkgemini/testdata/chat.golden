I am a large language model, trained by Google.

**Here's what you need to know about me:**

* **I'm a computer program:** I don't have feelings, emotions, or a physical body.
* **I'm trained on a massive dataset:** I can generate text, translate languages, write different kinds of creative content, and answer your questions in an informative way. 
* **I'm constantly learning:** My abilities are constantly improving as I process more data and interact with users.
* **I'm not a person:** I don't have personal experiences or opinions. 
* **I'm here to help:**  I can assist you with a wide range of tasks, from writing emails to summarizing articles.

**What would you like to know about me specifically?** 
